code/modules/overmap/ftl_shunt/core.dm

```diff
@@ -0,0 +1,551 @@
+	var/shunt_cancel_text = "Attention! Faster-than-light transition cancelled."
+	var/shunt_complete_text = "Attention! Faster-than-light transition completed."
+	var/shunt_spooling_text = "Attention! Superluminal shunt charge complete, spooling up."
+
+	var/shunt_sabotage_text_minor = "Warning! Electromagnetic flux beyond safety limits - aborting shunt!"
+	var/shunt_sabotage_text_major = "Warning! Critical electromagnetic flux in accelerator core! Dumping core and aborting shunt!"
+	var/shunt_sabotage_text_critical = "ALERT! Critical malfunction in microsingularity containment core! Safety systems offline!"
+
+	use_power = POWER_USE_OFF
+	power_channel = EQUIP
+	idle_power_usage = 1600
+	active_power_usage = 150000
+	icon_state = "bsd"
+	light_color = COLOR_BLUE
+
+//Base procs
+
+/obj/machinery/ftl_shunt/core/Initialize()
+	. = ..()
+	set_extension(src, /datum/extension/local_network_member)
+	if(initial_id_tag)
+		var/datum/extension/local_network_member/local_network = get_extension(src, /datum/extension/local_network_member)
+		local_network.set_tag(null, initial_id_tag)
+	find_ports()
+	set_light(1, 1, 2)
+
+/obj/machinery/ftl_shunt/core/Destroy()
+	. = ..()
+	for(var/obj/machinery/ftl_shunt/fuel_port/FP in fuel_ports)
+		FP.master = null
+		fuel_ports -= FP
+	ftl_computer.linked_core = null
+	ftl_computer = null
+
+/obj/machinery/ftl_shunt/core/on_update_icon()
+	overlays.Cut()
+
+	if(charging)
+		var/image/I = image('icons/obj/shunt_drive.dmi', "activating")
+		var/matrix/M = new()
+		I.transform = M
+		overlays += I
+
+	if(jumping)
+		overlays += image('icons/obj/shunt_drive.dmi', "activated")
+		var/image/S = image('icons/obj/objects.dmi', "bhole3")
+		var/matrix/M = new()
+		M.Scale(0.75)
+		S.transform = M
+		S.alpha = 0
+		animate(S, alpha = 255, time = 5.9 SECONDS)
+		overlays += S
+
+/obj/machinery/ftl_shunt/core/examine(mob/user)
+	. = ..()
+	if(sabotaged)
+		if(user.skill_check(SKILL_ENGINES, SKILL_ADEPT))
+			switch(sabotaged)
+				if(SHUNT_SABOTAGE_MINOR)
+					to_chat(user, SPAN_WARNING("It looks like it's been tampered with in some way, and the accelerator vanes seem out of place."))
+				if(SHUNT_SABOTAGE_MAJOR)
+					to_chat(user, SPAN_WARNING("Light behaves oddly around the core of [src], and it looks to have been tampered with! The vanes are definitely out of place."))
+				if(SHUNT_SABOTAGE_CRITICAL)
+					to_chat(user, SPAN_DANGER("Light bends around the core of [src] in a manner that eerily reminds you of a singularity... the vanes look completely misaligned!"))
+		else
+			to_chat(user, SPAN_WARNING("It looks like it's been tampered with, but you're not sure about to what extent."))
+
+/obj/machinery/ftl_shunt/core/attackby(var/obj/item/O, var/mob/user)
+	if(istype(O, /obj/item/stack/telecrystal))
+		var/obj/item/stack/telecrystal/TC = O
+		var/tc_input = input(user, "How many telecrystals do you want to put in?", "TC Input", 0) as num|null
+
+		if(!tc_input)
```

Comment:
Should check `user.incapacitated()`, `user.Adjacent(src)` and `TC in user.get_held_items()` post-input.